### PR TITLE
Add Flink Kafka Connector v3.0.0

### PR DESCRIPTION
**Pull request by tzulitai**

### CODE REVIEW

**Comment by asardaes** [comment]
If I may ask, is it right that this release will not be compatible with Flink 1.16.x?

File: docs/data/flink_connectors.yml [hunk 1]
```diff
@@ -50,6 +50,13 @@ jdbc:
   source_release_sha512_url: "https://downloads.apache.org/flink/flink-connector-jdbc-3.0.0/flink-connector-jdbc-3.0.0-src.tgz.sha512"
   compatibility: ["1.16.x"]
 
+kafka:
+  name: "Apache Flink Kafka Connector 3.0.0"
+  source_release_url: "https://www.apache.org/dyn/closer.lua/flink/flink-connector-kafka-3.0.0/flink-connector-kafka-3.0.0-src.tgz"
+  source_release_asc_url: "https://downloads.apache.org/flink/flink-connector-kafka-3.0.0/flink-connector-kafka-3.0.0-src.tgz.asc"
+  source_release_sha512_url: "https://downloads.apache.org/flink/flink-connector-kafka-3.0.0/flink-connector-kafka-3.0.0-src.tgz.sha512"
+  compatibility: ["1.17.x"]
+
```
Comment (MartijnVisser) [inline docs/data/flink_connectors.yml:59]: You also need to add this release to `docs/data/release_archive.yml`

**Review by MartijnVisser — APPROVED** [review]
LGTM

**Comment by tzulitai** [comment]
> If I may ask, is it right that this release will not be compatible with Flink 1.16.x?

@asardaes correct, this release is initially compatible with Flink 1.17.x only.

**Review by dannycranmer — APPROVED** [review]

**Review by vahmed-hamdy — APPROVED** [review]

**Pull request closed without merging**